File: backend/MovieScreening.Api/Contracts/Movie.cs
namespace MovieScreening.Api.Contracts;

public record Genre(string Id, string Name);
public record Movie(string Id, string Title, string OriginalTitle, int? Year,
    string Overview, int? Rating, string? PosterUrl, Genre[] Genres);
public record MoviePage(Movie[] Items, bool HasMore, string? NextCursor, string? Notice = null);

File: frontend/src/app/movie.ts
export interface Genre {
  id: string;
  name: string;
}
export interface Movie {
  id: string;
  title: string;
  originalTitle: string;
  year: number | null;
  overview: string;
  rating: number | null;
  posterUrl: string | null;
  genres: Genre[];
}
export interface MoviePage {
  items: Movie[];
  hasMore: boolean;
  nextCursor: string | null;
  notice: string | null;
}
export interface MovieSearch {
  title: string;
  year: number | null;
  genre: string;
  sortBy: "rating" | "title" | "year";
  direction: "asc" | "desc";
}
